### ISSUE TITLE
Questions about model load and model parameters, some parameters can not load rightly

### ISSUE THREAD

**Issue by Liujingxiu23** [pytorch/pytorch#58134]
My Model has two main sub models:
	self.model1 = SubModel1()
	self.model2 = SubModel2()
outputs = self.model2(self.model1(inputs))
	
Everythings works good util I want to use torch.jit.script to export the model.
The model can be exported successfully, but the weights of submodel2 is not right.

I tired the following code then:

model = Model(config)
model.load_state_dict(torch.load(checkpoint_path, map_location=torch.device('cpu'))['state_dict'] )
_ = model.eval()

print(" print weights of submodel1 the first time")
print(" print weights of submodel2 the first time")
scripted_module = torch.jit.script(model)
torch.jit.save(scripted_module, 'model-1.pt') # export the model the first time

outputs=model(inputs) # run inference

print(" print weights of submodel1 the second time") 
print(" print weights of submodel2 the second time") 
scripted_module = torch.jit.script(model)
torch.jit.save(scripted_module, 'model-2.pt') # export the model the second time

The results is:
The model-2.pt can do model inference right, while model-1.pt can not.
At the same time, the outputs of the two print operation of the submodel1 is the same, while the output of submodel2 is not the same and it seem the second time is right.

I wonder why this happend?

**Comment by Liujingxiu23** [comment]
My actual model used is https://github.com/LEEYOONHYUNG/BVAE-TTS.
And the submodel2 correspond to the "BVAE_Blocks"

The model mainly uses "conv1d".
The only layer which I seem special to me is "sn(Spectral Normalization)"

**Comment by Liujingxiu23** [comment]
Problem solved, cause the submodel2 has sn(torch.nn.utils.spectral_norm). 
"This is implemented via a hook that calculates spectral norm and rescales weight before every forward() call."
